### Heroes of Pymoli Data Analysis

In [1]:
# Dependencies and Setup
import pandas as pd

In [2]:
# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [3]:
purchase_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 780 entries, 0 to 779
Data columns (total 7 columns):
Purchase ID    780 non-null int64
SN             780 non-null object
Age            780 non-null int64
Gender         780 non-null object
Item ID        780 non-null int64
Item Name      780 non-null object
Price          780 non-null float64
dtypes: float64(1), int64(3), object(3)
memory usage: 42.8+ KB


In [4]:
purchase_data.columns

Index(['Purchase ID', 'SN', 'Age', 'Gender', 'Item ID', 'Item Name', 'Price'], dtype='object')

## Player Count

* Display the total number of players


In [5]:
player_demo = purchase_data.loc[:, ["Gender", "SN", "Age"]]
player_demo= player_demo.drop_duplicates()
num_player = player_demo.count()[0]
pd.DataFrame({"Total Number of Players":[num_player]})

Total Number of Players
0                      576

In [6]:
purchase_data.columns

Index(['Purchase ID', 'SN', 'Age', 'Gender', 'Item ID', 'Item Name', 'Price'], dtype='object')

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [7]:
purchase_df=purchase_data.copy()

In [8]:
# number of unique items
num_unique_items = len(purchase_df["Item Name"].unique())
# average purchase price
average_price = purchase_df["Price"].mean()

In [9]:
# number of purchase
numberofPurchase = len(purchase_df["Purchase ID"])
# Total revenue
Total_Rev = purchase_df["Price"].sum()
summary_df = pd.DataFrame({
    "Number of Unique Items": [num_unique_items],
    "Average Purchase Price": average_price,
    "Total Number of Purchases": numberofPurchase,
    "Total Revenue": Total_Rev
})
summary_df["Average Purchase Price"] = summary_df["Average Purchase Price"].map("${:.2f}".format)
summary_df["Total Revenue"] = summary_df["Total Revenue"].map("${:.2f}".format)
summary_df

Number of Unique Items Average Purchase Price  Total Number of Purchases  \
0                     179                  $3.05                        780   

  Total Revenue  
0      $2379.77

## Gender Demographics


* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [10]:
player_demo_total = player_demo["Gender"].value_counts()
player_demo_perc = player_demo_total/num_player *100
gender_demographics = pd.DataFrame({"Total Player": player_demo_total, "Percentage of Player": player_demo_perc})
gender_demographics["Percentage of Player"]= gender_demographics["Percentage of Player"].map("{:.2f}%".format)
gender_demographics


Total Player Percentage of Player
Male                            484               84.03%
Female                           81               14.06%
Other / Non-Disclosed            11                1.91%


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [11]:
gender_purchase_total = purchase_data.groupby(["Gender"]).sum()["Price"]
gender_purchase_total
gender_purchase_avg = purchase_data.groupby(["Gender"]).mean()['Price']
gender_counts = purchase_data.groupby(["Gender"]).count()["Price"]

#average total purchase per person
total_per_person = gender_purchase_total / gender_demographics["Total Player"]
#convert to dataFrame
gender_data_df = pd.DataFrame({"Purchase Count": gender_counts,
                               "Average purchase price": gender_purchase_avg,
                              "Total purchase value": gender_purchase_total,
                              "Average Purchase total per person by Gender": total_per_person})
gender_data_df["Average purchase price"]=gender_data_df["Average purchase price"].map("${:.2f}".format)
gender_data_df["Total purchase value"]=gender_data_df["Total purchase value"].map("${:.2f}".format)
gender_data_df["Average Purchase total per person by Gender"]=gender_data_df["Average Purchase total per person by Gender"].map("${:.2f}".format)                                                                                  
gender_data_df

Purchase Count Average purchase price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total purchase value  \
Gender                                       
Female                             $361.94   
Male                              $1967.64   
Other / Non-Disclosed               $50.19   

                      Average Purchase total per person by Gender  
Gender                                                             
Female                                                      $4.47  
Male                                                        $4.07  
Other / Non-Disclosed                                       $4.56

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [12]:
age_bins =[0, 9.5, 14.5, 19.5, 24.5, 29.5, 34.5, 39.5, 100]
group_names =["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]
player_demo["Age group"]=pd.cut(player_demo["Age"], age_bins, labels=group_names)

In [13]:
age_purchase_total = player_demo["Age group"].value_counts()
age_purchase_perc = age_purchase_total / num_player *100
age_purchase_df = pd.DataFrame({"Purchase counts": age_purchase_total,
        "Percentage of player": age_purchase_perc
})
age_purchase_df["Percentage of player"]=age_purchase_df["Percentage of player"].map("{:.2f}%".format)

In [14]:
age_purchase_df = age_purchase_df.sort_index()
age_purchase_df

Purchase counts Percentage of player
<10                 17                2.95%
10-14               22                3.82%
15-19              107               18.58%
20-24              258               44.79%
25-29               77               13.37%
30-34               52                9.03%
35-39               31                5.38%
40+                 12                2.08%

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [15]:
purchase_data["Age group"] = pd.cut(purchase_data["Age"], age_bins, labels=group_names)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age group  
0  Extraction, Quickblade Of Trembling Hands   3.53     20-24  
1                          Frenzied Scimitar   1.56       40+  
2                               Final Critic   4.88     20-24  
3                                Blindscythe   3.27     20-24  
4                                       Fury   1.44     20-24

In [16]:
total_purchase_by_age = purchase_data.groupby(["Age group"]).sum()["Price"]
age_average = purchase_data.groupby(["Age group"]).mean()["Price"]
age_counts = purchase_data.groupby(["Age group"]).count()["Price"]

In [17]:
avg_purchase_per_personbyAge = (total_purchase_by_age) / age_purchase_df["Purchase counts"]

In [18]:
age_data_df = pd.DataFrame({
    "Purchase COunt": age_counts, "Average purchase price": age_average,
    "Total purchase value": total_purchase_by_age,
    "Average Purchase Total per Person by AgeGroup": avg_purchase_per_personbyAge 
})
age_data_df["Average purchase price"] =age_data_df["Average purchase price"].map("${:.2f}".format)
age_data_df["Average Purchase Total per Person by AgeGroup"] =age_data_df["Average Purchase Total per Person by AgeGroup"].map("${:.2f}".format)

age_data_df

Purchase COunt Average purchase price  Total purchase value  \
Age group                                                                
<10                    23                  $3.35                 77.13   
10-14                  28                  $2.96                 82.78   
15-19                 136                  $3.04                412.89   
20-24                 365                  $3.05               1114.06   
25-29                 101                  $2.90                293.00   
30-34                  73                  $2.93                214.00   
35-39                  41                  $3.60                147.67   
40+                    13                  $2.94                 38.24   

          Average Purchase Total per Person by AgeGroup  
Age group                                                
<10                                               $4.54  
10-14                                             $3.76  
15-19                                             $3.86  
20-24                                             $4.32  
25-29                                             $3.81  
30-34                                             $4.12  
35-39                                             $4.76  
40+                                               $3.19

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [19]:
spender_total = purchase_data.groupby(["SN"]).sum()["Price"]
spender_average =purchase_data.groupby(["SN"]).mean()["Price"]
spender_count=purchase_data.groupby(["SN"]).count()["Price"]

In [20]:
spender_date_df = pd.DataFrame({"Total Purchase Value": spender_total,
                                "Average Purchase Value": spender_average,
                                "Purchase Count": spender_count
    })
#Sorted total purchase value
spender_date_sorted_df = spender_date_df.sort_values("Total Purchase Value", ascending=False)
spender_date_sorted_df = spender_date_sorted_df.loc[:, ["Purchase Count",  "Average Purchase Value", "Total Purchase Value"]]
spender_date_sorted_df["Average Purchase Value"] = spender_date_sorted_df["Average Purchase Value"].map("${:.2f}".format)
spender_date_sorted_df["Total Purchase Value"] = spender_date_sorted_df["Total Purchase Value"].map("${:.2f}".format)

In [21]:
spender_date_sorted_df.head()

Purchase Count Average Purchase Value Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [22]:
#Etract item data
item_data = purchase_data.loc[:,["Item ID", "Item Name", "Price"]]
# perform basic cal
total_purchase_item= item_data.groupby(["Item ID", "Item Name"]).sum()["Price"]
avg_purchase_item= item_data.groupby(["Item ID", "Item Name"]).mean()["Price"]
count_purchase_item= item_data.groupby(["Item ID", "Item Name"]).count()["Price"]


In [23]:
# Create new dataframe
item_df = pd.DataFrame({"Total Purchase Value":total_purchase_item, "Item Price": avg_purchase_item,
    "Purchase Count":count_purchase_item
})
item_sorted_df = item_df.sort_values("Purchase Count", ascending = False)
item_sorted_df["Total Purchase Value"]=item_sorted_df["Total Purchase Value"].map("${:.2f}".format)
item_sorted_df["Item Price"]=item_sorted_df["Item Price"].map("${:.2f}".format)
item_sorted_df.head()

Total Purchase Value  \
Item ID Item Name                                                           
92      Final Critic                                               $59.99   
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76   
145     Fiery Glass Crusader                                       $41.22   
132     Persuasion                                                 $28.99   
108     Extraction, Quickblade Of Trembling Hands                  $31.77   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                      Purchase Count  
Item ID Item Name                                                     
92      Final Critic                                              13  
178     Oathbreaker, Last Hope of the Breaking Storm              12  
145     Fiery Glass Crusader                                       9  
132     Persuasion                                                 9  
108     Extraction, Quickblade Of Trembling Hands                  9

In [24]:
item_popularity = item_sorted_df.loc[:, ["Purchase Count", "Item Price", "Total Purchase Value"]]
item_popularity.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [25]:
item_total_purchase = item_df.sort_values("Total Purchase Value", ascending = False)

item_profitable = item_total_purchase.loc[:, ["Purchase Count","Item Price","Total Purchase Value"]]
item_profitable["Item Price"]=item_profitable["Item Price"].map("${:.2f}".format)
item_profitable["Total Purchase Value"]=item_profitable["Total Purchase Value"].map("${:.2f}".format)
item_profitable.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

##  Three observable trends

1. The male players are the majority players which count for 84% of all players, female players count for 14%.
2. The average purchase total per person by gender has slight difference, with male of average of 4.07 and female of 4.47. 
3. The peak age group of player is between age 20 - 24 years old which counts for 45% of all the groups and the secondary groups falling between 15-19 (19%).